src/tests/suite/daos_dist_tx.c

```diff
@@ -2954,6 +2989,26 @@ dtx_39(void **state)
 	reintegrate_single_pool_rank(arg, kill_rank);
 }
 
+static void
+dtx_40(void **state)
+{
+	FAULT_INJECTION_REQUIRED();
+
+	print_message("DTX40: uncertain check - miss commit with delay\n");
+
+	dtx_uncertainty_miss_request(*state, DAOS_DTX_MISS_COMMIT, false, true);
+}
+
+static void
+dtx_41(void **state)
+{
+	FAULT_INJECTION_REQUIRED();
+
+	print_message("DTX41: uncertain check - miss abort with delay\n");
+
+	dtx_uncertainty_miss_request(*state, DAOS_DTX_MISS_ABORT, true, true);
+}
```

Comment:
the test only run for replica obj? is it convenient to also run it for EC obj? thanks

Reply:
Currently, it is only test replicated object. If we add EC object, this part will involve EC degraded fetch. I am not sure whether we can properly handle that if without the patch https://github.com/daos-stack/daos/pull/5967 or not. I would suggest to make another small patch to add new test case for EC object with hitting uncertain when EC degrade fetch becomes ready.

Reply:
Seems not directly related with PR5967, but we can add new test later as you said.